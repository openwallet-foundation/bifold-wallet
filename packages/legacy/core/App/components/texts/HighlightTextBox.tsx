import React from 'react'
import { StyleSheet, View } from 'react-native'

import { useTheme } from '../../contexts/theme'
import { ThemedText } from './ThemedText'

export interface TextBoxProps {
  children: string
}

const offset = 10

const HighlightTextBox: React.FC<TextBoxProps> = ({ children }) => {
  const { ColorPallet, OnboardingTheme } = useTheme()
  const style = StyleSheet.create({
    icon: {
      marginRight: offset,
    },
    container: {
      flexDirection: 'row',
      backgroundColor: ColorPallet.brand.primaryBackground,
    },
    accentBox: {
      marginRight: offset,
      backgroundColor: ColorPallet.brand.highlight,
      width: 8,
    },
    headerText: {
      ...OnboardingTheme.bodyText,
      flexShrink: 1,
    },
  })
  return (
    <View style={style.container}>
      <View style={style.accentBox} />
      <ThemedText style={[style.headerText, { paddingTop: offset, paddingBottom: offset }]}>{children}</ThemedText>
    </View>
  )
}

export default HighlightTextBox
