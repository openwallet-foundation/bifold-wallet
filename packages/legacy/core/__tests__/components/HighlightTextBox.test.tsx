import { render } from '@testing-library/react-native'
import React from 'react'

import HighlightTextBox from '../../App/components/texts/HighlightTextBox'

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')

describe('HighlightTextBox Component', () => {
  test('Renders correctly', () => {
    const tree = render(<HighlightTextBox>Hello World</HighlightTextBox>)

    expect(tree).toMatchSnapshot()
  })
})
