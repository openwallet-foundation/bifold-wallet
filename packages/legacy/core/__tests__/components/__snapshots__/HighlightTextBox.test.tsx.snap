// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightTextBox Component Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000000",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FCBA19",
        "marginRight": 10,
        "width": 8,
      }
    }
  />
  <Text
    maxFontSizeMultiplier={2}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "normal",
        },
        Array [
          Object {
            "color": "#FFFFFF",
            "flexShrink": 1,
            "fontSize": 18,
            "fontWeight": "normal",
          },
          Object {
            "paddingBottom": 10,
            "paddingTop": 10,
          },
        ],
      ]
    }
  >
    Hello World
  </Text>
</View>
`;
